tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -1125,6 +1160,13 @@ def tearDown(self):
             shutil.rmtree(self.device.storage_path, ignore_errors=True)
         self.apps = None
         self.data_layer = None
+
+        if self.capture != "off":
+            self.stop_video_capture()
+            # pull video file when there was an exception, or always set to pull
+            if self.capture == "always" or sys.exc_info()[0] is not None:
```

Comment:
Awesome! `sys.exc_info()[0]` is a great idea!

Just as a nit, I'd remove the comment and put the condition into a more self-explanatory function:
```py
def _has_thrown_any_exception_during_run(self):
    return sys.exc_info()[0] is not None
```